core-api/src/main/java/com/optimizely/ab/config/DatafileProjectConfig.java

```diff
@@ -480,7 +487,8 @@ public ProjectConfig build() throws ConfigParseException {
                 throw new ConfigParseException("Unable to parse empty datafile.");
             }
 
-            ProjectConfig projectConfig = DefaultConfigParser.getInstance().parseProjectConfig(datafile);
+            DatafileProjectConfig projectConfig = (DatafileProjectConfig) DefaultConfigParser.getInstance().parseProjectConfig(datafile);
```

Comment:
For safety, can we use "ProjectConfig" type and check instanceof before assigning datafile.
```suggestion
ProjectConfig projectConfig = DefaultConfigParser.getInstance().parseProjectConfig(datafile);
if (projectConfig instanceof DatafileProjectConfig) {
   ((DatafileProjectConfig)projectConfig).datafile = datafile;
```